futures-util/src/io/write_all_vectored.rs

```diff
@@ -10,21 +10,21 @@ use std::pin::Pin;
 /// [`write_all_vectored`](super::AsyncWriteExt::write_all_vectored) method.
 #[derive(Debug)]
 #[must_use = "futures do nothing unless you `.await` or poll them"]
-pub struct WriteAllVectored<'a, W: ?Sized + Unpin> {
+pub struct WriteAllVectored<'a, 'b, W: ?Sized + Unpin> {
     writer: &'a mut W,
-    bufs: &'a mut [IoSlice<'a>],
+    bufs: &'a mut [IoSlice<'b>],
 }
 
-impl<W: ?Sized + Unpin> Unpin for WriteAllVectored<'_, W> {}
+impl<W: ?Sized + Unpin> Unpin for WriteAllVectored<'_, '_, W> {}
 
-impl<'a, W: AsyncWrite + ?Sized + Unpin> WriteAllVectored<'a, W> {
-    pub(super) fn new(writer: &'a mut W, mut bufs: &'a mut [IoSlice<'a>]) -> Self {
+impl<'a, 'b, W: AsyncWrite + ?Sized + Unpin> WriteAllVectored<'a, 'b, W> {
+    pub(super) fn new(writer: &'a mut W, mut bufs: &'a mut [IoSlice<'b>]) -> Self {
         IoSlice::advance_slices(&mut bufs, 0);
         Self { writer, bufs }
     }
 }
 
-impl<W: AsyncWrite + ?Sized + Unpin> Future for WriteAllVectored<'_, W> {
+impl<W: AsyncWrite + ?Sized + Unpin> Future for WriteAllVectored<'_, '_, W> {
```

Comment:
Isn't this `impl<'a, W> ... WriteAllVectored<'a, 'a, W> {}`?

Reply:
Would you be able to elaborate on why you think so? AIUI `Future` should be valid for any choice of lifetimes, and it doesn't need to refer to them by name.

Reply:
Nvm, was mistaken.